adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralNativeAdLoader.m

```diff
@@ -0,0 +1,245 @@
+    _adConfiguration = adConfiguration;
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationNativeLoadCompletionHandler originalCompletionHandler =
+        [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationNativeAdEventDelegate>(
+        _Nullable id<GADMediationNativeAd> ad, NSError *_Nullable error) {
+      if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+        return nil;
+      }
+      id<GADMediationNativeAdEventDelegate> delegate = nil;
+      if (originalCompletionHandler) {
+        delegate = originalCompletionHandler(ad, error);
+      }
+      originalCompletionHandler = nil;
+      return delegate;
+    };
+    
+    UIViewController *rootViewController = adConfiguration.topViewController;
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    if (!adUnitId.length || !placementId.length) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID or Placement ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    _adUnitId = adUnitId;
+    _nativeManager = [[MTGBidNativeAdManager alloc]initWithPlacementId:placementId unitID:adUnitId presentingViewController:rootViewController];
+    _nativeManager.delegate = self;
+    [_nativeManager loadWithBidToken:adConfiguration.bidResponse];
+}
+
+- (MTGMediaView *)createMediaView {
+    if (_mediaView) {
+        return _mediaView;
+    }
+    _mediaView= [[MTGMediaView alloc] initWithFrame:CGRectZero];
+    _mediaView.delegate = self;
+    return _mediaView;
+}
+
+#pragma mark - MTGBidNativeAdManagerDelegate
+- (void)nativeAdsLoaded:(nullable NSArray *)nativeAds bidNativeManager:(nonnull MTGBidNativeAdManager *)bidNativeManager {
+    
+    if ([nativeAds isKindOfClass:NSArray.class] && nativeAds.count > 0) {
+        _campaign = nativeAds.firstObject;
+        
+        _icon = [self loadImageWithURLString:_campaign.iconUrl];
+        GADNativeAdImage *image = [self loadImageWithURLString:_campaign.imageUrl];
+        if (image) {
+            _images = @[image];
+        }
+        
+        MTGMediaView *mediaView = [self createMediaView];
+        GADMAdapterMintegralExtras *extras = _adConfiguration.extras;
+        if(extras){
+            mediaView.mute = extras.muteVideoAudio;
+        }
+
+        if (_adLoadCompletionHandler) {
+            _adEventDelegate = _adLoadCompletionHandler(self,nil);
+        }
+
+    }else{
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorAdNotAvailable, @"No native ad returned from Mintegral.");
```

Comment:
Nit, can you update the error message to: `Mintegral SDK failed to return a native ad.`?

Reply:
OK, I've made changes and committed.